zeebe/snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/FileBasedSnapshotStore.java

```diff
@@ -77,9 +78,16 @@ public final class FileBasedSnapshotStore extends Actor
   private final AtomicLong receivingSnapshotStartCount;
   private final Set<PersistableSnapshot> pendingSnapshots = new HashSet<>();
   private final Set<FileBasedSnapshot> availableSnapshots = new HashSet<>();
+  private ChecksumProvider checksumProvider;
   private final String actorName;
   private final int partitionId;
 
+  public FileBasedSnapshotStore(
+      final int partitionId, final Path root, final ChecksumProvider checksumProvider) {
+    this(partitionId, root);
+    this.checksumProvider = checksumProvider;
+  }
+
```

Comment:
:wrench: Same thing with the constructors here. It's easier to understand if there's one full constructor that all others delegate to.